Doc/library/tarfile.rst

```diff
@@ -100,7 +100,10 @@ Some facts and figures:
 
    For modes ``'w:gz'``, ``'r:gz'``, ``'w:bz2'``, ``'r:bz2'``, ``'x:gz'``,
    ``'x:bz2'``, :func:`tarfile.open` accepts the keyword argument
-   *compresslevel* (default ``9``) to specify the compression level of the file.
+   *compresslevel* to specify the compression level of the file.
+   This defaults to ``9`` for gzip and bz2 compression and to
+   :class:`PRESET_DEFAULT <lzma.LZMACompressor>` for lzma compression.
+   *compresslevel* is ignored for other modes.
```

Comment:
It looks like you want modes like _w:xz_ to respect _compresslevel_, rather than ignore it.

It would be good to clarify if “other modes” includes the “special purpose” streaming modes like _w|xz_ introduced below.

Reply:
Hmm, the streaming modes fail with an exception as well.

I see now that the docs are correct - the listed modes (and only the listed modes) accept `compresslevel`.

Perhaps a better PR would just document that `preset` works for `w:xz`. This is totally undocumented so far.